packages/block-library/src/table-of-contents/index.php

```diff
@@ -0,0 +1,174 @@
+<?php
+/**
+ * Server-side rendering of the `core/table-of-contents` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Registers the `core/table-of-contents` block on the server.
+ *
+ * @since 6.4.0
+ */
+function register_block_core_table_of_contents() {
+	// @todo: Register meta for all post types that use Block Editor.
+	foreach ( array( 'post', 'page' ) as $post_type ) {
```

Comment:
To register it for all post types that support the block editor, we can do something like:

```suggestion
	$post_types = get_post_types( [
		'supports' => 'editor',
		'show_in_rest' => true,
	] );
	foreach ( $post_types as $post_type ) {
```
```

Reply:
We can "query" the `get_post_types` based on the `supports` array. This is the next best thing I can come up with.

```php
$supported_post_types = get_post_types(
  array(
	  'public'       => true,
	  'show_in_rest' => true,
  )
);
```

Reply:
Adding `public => true` to `supports => editor` and `show_in_rest => true` makes sense 👍

Reply:
Sorry, I had a typo. We can't query based on `supports` property arguments. If you run code with it will return an empty array.

Reply:
Ah, now I understand what you meant 👍  

I guess we can go with the `get_post_types()` "query", with `public` and `show_in_rest`, and then in the subsequent `foreach()` do something like:

```
foreach ( $post_types as $post_type ) {
	if ( ! post_type_supports( $post_type, 'editor' ) ) {
		continue;
	}
	// ...
}
```

What do you think?

Reply:
Good idea!

Reply:
Seems like it's a known issue, and old one at that, and not much has changed in that area: https://core.trac.wordpress.org/ticket/17620

Reply:
Fixes in [2e549dc](https://github.com/WordPress/gutenberg/pull/54224/commits/2e549dc88378f5887868ac711e4510de460a856c).